src/test/java/org/apache/maven/plugins/dependency/AbstractDependencyMojoTestCase.java

```diff
@@ -35,33 +36,52 @@
 import org.eclipse.aether.RepositorySystemSession;
 import org.eclipse.aether.repository.LocalRepository;
 import org.eclipse.aether.repository.LocalRepositoryManager;
+import org.junit.Before;
 import org.sonatype.plexus.build.incremental.DefaultBuildContext;
 
 public abstract class AbstractDependencyMojoTestCase extends AbstractMojoTestCase {
 
     protected File testDir;
-
     protected DependencyArtifactStubFactory stubFactory;
 
-    protected void setUp(String testDirectoryName, boolean createFiles) throws Exception {
-        setUp(testDirectoryName, createFiles, true);
-    }
-
-    protected void setUp(String testDirectoryName, boolean createFiles, boolean flattenedPath) throws Exception {
-        // required for mojo lookups to work
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        // Initialize mojo lookups as required by AbstractMojoTestCase
         super.setUp();
 
+        // Create a unique temporary test directory to avoid parallel test conflicts
+        String uniqueDirName = "test-dependency" + UUID.randomUUID();
```

Comment:
I don't think you need UUIDs here. JUnit can create temporary directories with @TempDir. Or do we not have JUnit 5 here? Either way, let's makes sure we only create one temporary directory per test.

Reply:
> JUnit can create temporary directories with `@Tempdir`.

I'm not sure `@Tempdir` would work here, for the same reason as in https://github.com/apache/maven-dependency-plugin/pull/532/files#r2115815689.

> Or do we not have JUnit 5 here?

It's not enough in test-time.